test/lib/componentManager.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  PermissionDialog,
+  SNComponent,
+  SNComponentManager,
+  SNTheme,
+} from '@standardnotes/snjs';
+
+export class WebComponentManager extends SNComponentManager {
```

Comment:
one class per file - there are multiple classes in this file

Reply:
I don't understand why? These two classes are related. If you change SNComponentManager to add another override chances are you'll want to take a look at both implementations. One could argue that you can achieve this with a folder structure but then that really just highlights that it's an arbitrary decision. I don't think it's damaging at all to have multiple classes in one file.

Reply:
Using one class per file gives a clear picture of the code's abstract structure. Hiding multiple classes in single files hides the complexity of the design implemented. 

Example 1 - if you look at this folder structure: https://github.com/standardnotes/syncing-server-js/tree/develop/src/Domain
you can clearly see what are the key domain factors in this application

Example 2 - if you look at this folder structure: https://github.com/standardnotes/syncing-server-js/tree/develop/src/Domain/UseCase you can clearly see what are the use cases of the application and what it can do.

Hiding code complexity in single files is never a good thing. Writing tests for those classes also becomes a mess.

Reply:
I've personally also leaned towards one class per file, probably solely because sometimes I'd open a file and just expect that there is one class in there, and end up not being able to find a class I was looking for because I had to scroll down.

Also, files are typically named after the class. But if you have multiple classes, how do you name the file? There isn't a clear paradigm there.

Reply:
> one class per file - there are multiple classes in this file

I agree on this. But in this case, these two classes are used to aid in testing, not really part of the main code.

Both classes are on the same file for convenience and because they are related.

Reply:
my bad - as part of test setup it's ok